Source/utils/format_int.cpp

```diff
@@ -0,0 +1,39 @@
+#include "utils/format_int.hpp"
+
+#include <fmt/compile.h>
+#include <fmt/core.h>
+
+#include "utils/language.h"
+#include "utils/stdcompat/string_view.hpp"
+
+namespace devilution {
+
+std::string FormatInteger(int n)
+{
+	constexpr size_t GroupSize = 3;
+
+	char buf[40];
+	const char *begin = buf;
+	size_t len = fmt::format_to(buf, FMT_COMPILE("{}"), n) - begin;
+
+	std::string out;
+	out.reserve(len + (len - 1) / GroupSize + (n < 0 ? 1 : 0));

```

Comment:
Should this account for translated contexts where the separator is a multibyte character (or string...)?

Reply:
Oh right, will fix

Reply:
Fixed along with other improvements